lib/internal/streams/destroy.js

```diff
@@ -331,7 +333,12 @@ function destroyer(stream, err) {
     return;
   }
 
-  if (!err && !isFinished(stream)) {
+  // Readable.destroy(null) is explicit end of stream.
+  if (err === null) {
```

Comment:
So this is a little tricky... there is actually a valid case for calling `Readable.destroy()` without an error.... when you only want to read a limited amount of elements, e.g. `take(5)` which shouldn't really end up as an abort error.

To keep readable/writable semantics the same I also added the `stream.end()` logic here. Not strictly necessary.

Reply:
Another option is to add a method to `Readable`, e.g. `Readable.end()` to signal "not more data needed and it's not an error`.

Reply:
@mcollina @lpinca